// react-sensei/dist/navs/sidebars/clean/NavMenu.js
import React from "react";
import styled, { css } from "styled-components";
const StyledWrapper = styled.div`
  padding: ${p => p.theme.lenMd3} 0 0;
`;
const activeStyles = css`
  padding: 0 ${p => p.theme.lenMd1};
  box-shadow: ${p => p.theme.boxshadowMd};
  margin-bottom: ${p => p.theme.lenMd2};
  border-radius: ${p => p.theme.lenLg2};
  background: ${p => p.theme.colorPrimary};
  color: ${p => p.theme.colorPrimaryText};
`;
const StyledLink = styled.button`
  display: block;
  font-weight: ${p => p.theme.fontWeightBold};
  width: 100%;
  height: ${p => p.theme.lenLg2};
  color: ${p => p.theme.colorText};

  ${p => p.isActive && activeStyles};
`;

const Wrapper = () => {
  return /*#__PURE__*/React.createElement(StyledWrapper, null, /*#__PURE__*/React.createElement(StyledLink, {
    isActive: true
  }, "Dashboard"), /*#__PURE__*/React.createElement(StyledLink, null, "Analysis"), /*#__PURE__*/React.createElement(StyledLink, null, "Messages"), /*#__PURE__*/React.createElement(StyledLink, null, "Invoices"), /*#__PURE__*/React.createElement(StyledLink, null, "Payslips"));
};

export default Wrapper;
