// react-sensei/dist/navs/sidebars/clean/ProfileBlock.js
import React from "react";
import styled from "styled-components";
const StyledTopbar = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  padding: ${p => p.theme.lenMd2};
  margin-bottom: ${p => p.theme.lenMd1} ${p => p.theme.lenSm2} 0;
`;
const StyledProfileImage = styled.img`
  width: 140px;
  height: 140px;
  border-radius: 50%;
  margin-bottom: ${p => p.theme.lenLg2};
  box-shadow: ${p => p.theme.boxshadowLg};
`;
const StyledProfileName = styled.div`
  margin-bottom: ${p => p.theme.lenSm2};
  font-size: ${p => p.theme.lenMd3};
`;
const StyledProfileRole = styled.div`
  color: ${p => p.theme.colorLabel};
  font-size: ${p => p.theme.lenMd1};
  padding-bottom: ${p => p.theme.lenLg1};
  border-bottom: 2px solid ${p => p.theme.colorBorder};
`;
const StyledTopbarIcon = styled.span`
  color: ${p => p.theme.colorText};
  font-size: ${p => p.theme.lenLg1} !important;
`;

const Wrapper = ({
  profileImageUrl
}) => {
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(StyledTopbar, null, /*#__PURE__*/React.createElement(StyledTopbarIcon, {
    className: "material-icons"
  }, "menu"), /*#__PURE__*/React.createElement(StyledTopbarIcon, {
    className: "material-icons"
  }, "settings")), /*#__PURE__*/React.createElement(StyledProfileImage, {
    src: profileImageUrl
  }), /*#__PURE__*/React.createElement(StyledProfileName, null, "Virat Kohli"), /*#__PURE__*/React.createElement(StyledProfileRole, null, "Top order batsman"));
};

export default Wrapper;
