// react-sensei/dist/cards/charts/gradientLineChart/GradientLineWrapper.js
import React, { useState, useEffect } from "react";
import styled from "styled-components";
import { useStyledTheme } from "../../../hooks";
import GradientLineChart from "./GradientLineChart";
import { InfoStack } from "../../../";
import { CardFixedHeight } from "../shared";
const StyledStatList = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-around;
  width: 100%;
`;
const StyledInfoStack = styled(InfoStack)`
  flex: 1 1 33.33%;
`;

const Wrapper = ({
  cardTitle,
  statList,
  data
}) => {
  const theme = useStyledTheme();
  const [prevThemeId, setPrevThemeId] = useState(undefined);
  const [themeLoaded, setThemeLoaded] = useState(false);
  useEffect(() => {
    console.log("themeChanged", themeLoaded);
    setThemeLoaded(false);

    if (theme.id !== prevThemeId) {
      setTimeout(() => {
        setThemeLoaded(true);
      }, 10);
    }
  }, [theme.id]);
  return /*#__PURE__*/React.createElement(CardFixedHeight, {
    cardTitle: cardTitle
  }, statList && /*#__PURE__*/React.createElement(StyledStatList, null, statList.map(stat => /*#__PURE__*/React.createElement(StyledInfoStack, {
    value: stat.value,
    label: stat.label
  }))), themeLoaded ? /*#__PURE__*/React.createElement(GradientLineChart, {
    cardTitle: cardTitle,
    statList: statList,
    data: data
  }) : null);
};

export default Wrapper;
